pages/spots/index.js

```diff
@@ -0,0 +1,36 @@
+import styled from "styled-components";
+import SpotCard from "../../components/SpotCard.js";
+
+export default function Spots({ spots, setSpots }) {
+  function toggleBookmark(id) {
+    const newSpotArray = spots.map((spot) => {
+      if (spot.id === id) {
+        return {
+          ...spot,
+          isBookmark: !spot.isBookmark,
+        };
+      } else {
+        return spot;
+      }
+    });
+    setSpots(newSpotArray);
+  }
```

Comment:
you can use the toggleBookmark function you are passing down and don't need to redefine it here:

```suggestion
export default function Spots({ spots, toggleBookmark }) {
```